src/DotLiquid/Context.cs

```diff
@@ -105,6 +103,28 @@ public Context
              , int maxIterations
              , IFormatProvider formatProvider
              , CancellationToken cancellationToken)
+            : this(environments, outerScope,registers, errorsOutputMode, maxIterations, null, formatProvider, new List<Exception>(), cancellationToken)
+        {
+        }
+
+        /// <summary>
+        /// Creates a new rendering context
+        /// </summary>
+        public Context(IFormatProvider formatProvider)
+            : this(new List<Hash>(), new Hash(), new Hash(), ErrorsOutputMode.Display, 0, 0, formatProvider)
+        {
+        }
+
+        internal Context
+            (List<Hash> environments
+             , Hash outerScope
+             , Hash registers
+             , ErrorsOutputMode errorsOutputMode
+             , int maxIterations
+             , int? timeout

```

Comment:
all constructors with timeout are marked obsolete. See our other PR discussion about this. We should not be adding new constructors that include obsolete parameters. Let's hold until the other PR is worked out.